src/Microsoft.ML.Data/DataLoadSave/EstimatorChain.cs

```diff
@@ -72,7 +72,7 @@ public EstimatorChain<TNewTrans> Append<TNewTrans>(IEstimator<TNewTrans> estimat
             where TNewTrans : class, ITransformer
         {
             Contracts.CheckValue(estimator, nameof(estimator));
-            return new EstimatorChain<TNewTrans>(_estimators.Append(estimator).ToArray(), _scopes.Append(scope).ToArray());
+            return new EstimatorChain<TNewTrans>(_estimators.Concat(new[] { estimator }).ToArray(), _scopes.Concat(new[] { scope }).ToArray());
```

Comment:
I'm not sure how concerned we are about perf in these methods, but another way would be to make our own `AppendElement` method:

```C#
public static T[] AppendElement(T[] array, T element)
{
    T[] result = new T[array.Length + 1];
    Array.Copy(array, result);
    result[array.Length] = element;
    return result;
}
```

Another option would be to use ImmutableCollections.... I don't think you'd have to pay a penalty for copying in that case. But I'm not sure we'd want to take a new dependency, if we aren't using ImmutableCollections already.

Reply:
Hi @eerhardt -- we are using `ImmutableCollections`, and I am beginning to use it in my code. I really like it a lot, I think we ought to use it more.